main/guides/js-programming/notifiers.md

```diff
@@ -1,94 +1,330 @@
+to all subscribers, `SubscriptionKit` generally should not be used with rapidly
+produced values.
+
+An iteration’s suffix subset is defined by its starting point in the original
+iteration, which can be a non-final value or a termination.
+The suffix subset has exactly the original iteration’s members from its starting
+point to and including its termination. For example, if the original is
+{ 2 5 9 13 Fail } with Fail as the termination and a starting point at 9, the
+subset is { 9 13 Fail }.
+
+When a new value becomes available on the original iteration, it promptly becomes
+available on every suffix subset whose starting point is at or before that value
+So if the original is { 2 5 9 13 Fail } and 9 becomes available, 9 promptly
+becomes available to any suffix subset with a starting point of 2, 5, or 9. It
+does not become available to any subset starting at 13 or Fail).
+
+Each subscription is an `AsyncIterable` that produces any number of
+`AsyncIterators`. These `AsyncIterators` are `SubsciptionIterators` which also
+have a `subscribe()` method. Calling a `subscribe()` method makes a
+`Subscription` whose starting point is that `SubscriptionIterator`'s current
+position.
+
+## Methods
+
+The `updater` and `publication` both have the same three methods:
+- `updateState(state)`
+  Supplies and sends out a new state to consumers. All active Promises 
+  produced by `getUpdateSince()` are resolved to the next record.
+- `finish(finalState)`
+  - Closes the stream of state changes and supplies a final state
+    value to consumers.
+- `fail(reason)`
+  - Closes the stream of state changes, indicates a failure to finish
+    satisfactorily, and supplies a reason for the failure to consumers. Does not provide
+    a next state. Instead, it causes the Promise to be rejected with the reason, 
+    signalling that the monitored object hit an error condition.
+
+The `subscription` and `publication` both have this method:
+- `getUpdateSince()`: Returns `{ value, updateCount }`. 
+  - Returns the next published value after the previously obtained value.
+    `value` represents the state, and the format is up to the publisher.
+    `updateCount` requests notification the next time there's a state change.
+    If the state becomes final (e.g. a seat exits), `updateCount` will be 
+    undefined. If there's an error, the promise for the record is 
+    rejected and there isn't a next state.
+
+    If you call `getUpdateSince(oldUpdateCount)` with no count, or any 
+    `updateCount` other than the most recent one, the notifier immediately 
+    returns a promise for a record with the current state. If you call with 
+    the most-recently generated `updateCount`, the notifier returns a promise 
+    for the next record, which is resolved on the next state change. If you 
+    haven't called `getUpdateSince()` before, you won't have a 
+    previous `updateCount` to use.
+
+## Notifiers and Subscriptions in Zoe
```

Comment:
My review ends at Notifiers and Subscriptions in Zoe. Everything above looks _fantastic_.